File: NFMWorld.Library/Mad/IStage.cs
using nfm_world_library.Lua;
using NFMWorldLibrary.Collision;
using NFMWorldLibrary.FixedMath;

namespace NFMWorldLibrary;

[LuaVisible]
public partial interface IStage
{
    [LuaHidden]
    ReadOnlySpan<CollisionShapeRef> RetrievePointCollidables(fix64 x, fix64 z);
    IReadOnlyList<ITransform> pieces { get; }
    IReadOnlyList<IAiNode> nodes { get; }
    IReadOnlyList<IAiNode> checkpoints { get; }
    IReadOnlyList<IAiNode> fixHoops { get; }
    ushort nlaps { get; }
    ITransform CreateObject(string objectName, int x, int y, int z, int xz); 
}

File: NFMWorld.Library/data/lua/library/IStage.lua
---@class IStage
---@field pieces System.Collections.Generic.IReadOnlyList_ITransform
---@field nodes System.Collections.Generic.IReadOnlyList_IAiNode
---@field checkpoints System.Collections.Generic.IReadOnlyList_IAiNode
---@field fixHoops System.Collections.Generic.IReadOnlyList_IAiNode
---@field nlaps integer
---@field createObject fun(self: IStage, objectName: string, x: integer, y: integer, z: integer, xz: integer): ITransform

IStage = {}

